libraries/standard/mqtt/include/types/iot_mqtt_common_types.h

```diff
@@ -0,0 +1,557 @@
+/*
+ * IoT MQTT V2.1.0
```

Comment:
I do not think we should split the types file.  It seems like types in this header will be different based on whether the lightweight or full API is used, but retain the same name across both.  It is confusing to have the same name refer to different types.

I would prefer all types in 1 file.  The lightweight API should just ignore the full API's types.

Reply:
The common types from iot_mqtt_types.h will be removed and iot_mqtt_types.h will include iot_mqtt_common_types.h.  The reason I want to separate those two is to avoid including taskpool and other data types not needed by LW API. Also I'm not creating separate data types for LW API to avoid code duplication.